### PR TITLE
Community article for IoT Core gateways tutorial

### PR DESCRIPTION
**Pull request by hongalex**
- All images exist in the `images/` folder for now, and are relatively referenced in `index.md`

### CODE REVIEW

**Comment by ToddKopriva** [comment]
- [ ] editorial review: @ToddKopriva 
- [ ] final production review for publishing: @ToddKopriva

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+## Enable Cloud IoT Core and Cloud Pub/Sub APIs
+
+Click **Enable API** for each of the following:
+
+- [Cloud IoT Core][cloud-iot] (for managing gateways and devices)
+- [Cloud Pub/Sub](http://console.developers.google.com/cloudpubsub) (for ingesting device telemetry)
+
+[cloud-iot]: https://console.developers.google.com/iot
+
+## Create a Registry
```
Comment (ToddKopriva) [inline tutorials/cloud-iot-gateways-rpi/index.md]: 'Create a Registry'
->
'Create a registry'

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+First, create a device registry that will contain your gateway and devices.
+
+1. Open the [Cloud IoT Core console][cloud-iot].
+2. Ensure the right project is selected in the upper left.
+3. Click on **Create a device registry**.
+4. For **Registry ID**, enter `my-registry`.
+5. Select the region closest to you, but for the purposes of this demo we'll use `us-central1`.
+6. Ensure the MQTT protocol is enabled for this registry.
+7. Under **Cloud Pub/Sub topics**, create a new Cloud Pub/Sub topic for **Default telemetry topic**.
+    - Click on the **Select a Cloud Pub/Sub topic** Dropdown.
```
Comment (ToddKopriva) [inline tutorials/cloud-iot-gateways-rpi/index.md]: No need to capitalize 'dropdown' here.

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    - Click on the **Create a topic** option.
+    - Enter a PubSub topic name, such as `gateway-telemetry`.
```
Comment (ToddKopriva) [inline tutorials/cloud-iot-gateways-rpi/index.md]: 'PubSub'
->
'Cloud Pub/Sub'

https://cloud.google.com/guides/style/branding-guidelines

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+8. Do the same for **Device state topic**, under a different PubSub topic named `gateway-state`.
+
+    ![create registry][create-registry]
+
+9. Leave everything else as-is, and click **Create**.
+
+[create-registry]: ./images/create-registry.png 
+
+## Set Up your Gateway
```
Comment (ToddKopriva) [inline tutorials/cloud-iot-gateways-rpi/index.md]: I'll stop marking the headings, but all should be in sentence case.

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+14. Find the local IP address of the gateway using `ifconfig` on  Mac/Linux or `ipconfig /all` on Windows. Copy this somewhere as you will need to add this IP address to `led-light.py` and `thermostat.py` later.
+
+[extra-notes]: #extra-notes
+[create-gateway]: ./images/create-gateway.png
+
+## Raspberry Pi Setup
+
+In this tutorial, you'll use a [Raspberry Pi][rpi] to manage the LED/temperature sensor. Devices will connect to the gateway device through [UDP sockets][udp-socket], which will connect to Cloud IoT Core via the [MQTT bridge][mqtt-bridge]. A Raspberry Pi could theoretically connect directly to the cloud (since the Pi can connect to the internet), so using a Raspberry Pi for this part is purely for demonstration purposes.
+
+1. [Download Raspian][raspbian-download] (the full image with Desktop and recommended software) and follow [the installation guide][raspbian-installation] to flash Raspbian onto your microSD card.
```
Comment (ToddKopriva) [inline tutorials/cloud-iot-gateways-rpi/index.md]: 'Raspian'
->
'Raspbian'

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+2. Insert the microSD card with Raspbian into your Raspberry Pi.
+3. Attach a power source to the Raspberry Pi using the microUSB cable (i.e. to a laptop USB port).
```
Comment (ToddKopriva) [inline tutorials/cloud-iot-gateways-rpi/index.md]: Do you mean 'e.g.' (which means "for example") or do you mean 'i.e.' (which means "that is")?

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+8. Clone the following repository and change into the directory for this tutorial's code:
+
+        git clone https://github.com/GoogleCloudPlatform/community.git
+        cd community/tutorials/cloud-iot-gateways-pi
+
+9. Install Python dependencies by running the following:
+
+        pip install -r requirements.txt
+
+    Feel free to do this part in the virtual environment manager of your choosing, though make sure to switch into those environments when opening new tabs
```
Comment (ToddKopriva) [inline tutorials/cloud-iot-gateways-rpi/index.md]: missing period at end of sentence

File: tutorials/cloud-iot-gateways-rpi/index.md [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+description: Learn how to setup gateways on Cloud IoT Core using a Raspberry Pi.
+author: hongalex
+tags: Cloud IoT Core, Gateways, Raspberry Pi, Python, MQTT, internet of things
+date_published: 2018-12-10
+---
+
+Alex Hong | Developer Programs Engineer | Google Cloud IoT Core  
+Fengrui Gu | Software Engineer | Google Cloud IoT Core
+
+This tutorial shows you how to set up and use gateways on Cloud IoT Core. From the [documentation][gateways-overview], a "gateway is a device that connects less capable devices to Cloud IoT Core and performs several tasks on the device's behalf, such as communication, authentication, storage, and processing."
```
[the remark this answers is not in the document]
Comment (hongalex) [reply tutorials/cloud-iot-gateways-rpi/index.md]: I tried to use reference style links for either of the following cases:
1. If the link appeared multiple times in the document
2. If I wanted to to find and replace links that needed to be updated often (due to documentation link changing, images, etc.)

File: tutorials/cloud-iot-gateways-rpi/index.md (hunk 4 above)
Comment (hongalex) [reply tutorials/cloud-iot-gateways-rpi/index.md]: Changed the casing for the other headings

File: tutorials/cloud-iot-gateways-rpi/index.md (hunk 6 above)
Comment (hongalex) [reply tutorials/cloud-iot-gateways-rpi/index.md]: Whoops, yes, will change to 'e.g'

**Comment by ToddKopriva** [comment]
published here:
https://cloud.google.com/community/tutorials/cloud-iot-gateways-rpi

**Pull request merged**